src/mochiweb_headers.erl

```diff
@@ -215,6 +215,9 @@ tokenize_header_value(undefined) ->
 tokenize_header_value(V) ->
     reversed_tokens(trim_and_reverse(V, false), [], []).
 
+trim_leading_and_trailing_ws(S) ->
+    re:replace(S, "^[ \\t]*|[ \\t]*$", "", [global, {return, list}]).
```

Comment:
suggest `string:trim(String, both, [9,32]).` for performance;

```RE input: '     hello there     ', output: 'hello there', speed: 6.276318
TRIM input: '     hello there     ', output: 'hello there', speed: 1.377677```

```benchmark(F) ->
    Count = 1_000_000,
    {Time, _} = timer:tc(fun() -> lists:foreach(fun(_) -> F() end, lists:seq(1, Count)) end),
    Time / Count.```

Reply:
I had the same idea too but then noticed mochiweb supports Erlang 18+ [versions](https://github.com/mochi/mochiweb/blob/main/rebar.config#L3) and `string:trim/1` is an Erlang 20+ [function](https://www.erlang.org/doc/man/string.html#trim-1).

Reply:
> suggest `string:trim(String, both, [9,32]).` for performance;
> 
> ```
> RE input: '     hello there     ', output: 'hello there', speed: 6.276318
> TRIM input: '     hello there     ', output: 'hello there', speed: 1.377677
> ```
Thats significant. Can we make a runtime distinction and differentiate between the Erlang versions? So if Erlang > 20 use `trim/3` else use `strip/3`?

> I had the same idea too but then noticed mochiweb supports Erlang 18+ [versions](https://github.com/mochi/mochiweb/blob/main/rebar.config#L3) and `string:trim/1` is an Erlang 20+ [function](https://www.erlang.org/doc/man/string.html#trim-1).
If we want to be RFC compliant, we can't use `trim/1` because that would remove more chars than `space` and `tab`.
```
OWS = *( SP / HTAB )
    ; optional whitespace
```